File: android/app/src/main/java/tech/acab/app/ble/AcabProfile.kt
package tech.acab.app.ble

import java.util.UUID

/** The ACAB GATT contract. Keep in sync with firmware/src/oui-spy/acab_ble_service.h. */
object AcabProfile {
    val SERVICE: UUID    = UUID.fromString("acab0100-6f75-6973-7079-000000000000")
    val DETECTIONS: UUID = UUID.fromString("acab0101-6f75-6973-7079-000000000000") // notify (encrypted)
    val CONFIG: UUID     = UUID.fromString("acab0102-6f75-6973-7079-000000000000") // write  (encrypted)
    val STATUS: UUID     = UUID.fromString("acab0103-6f75-6973-7079-000000000000") // read + notify (encrypted)

    /** Standard CCCD descriptor — the one you write to turn notifications on. */
    val CCCD: UUID       = UUID.fromString("00002902-0000-1000-8000-00805f9b34fb")
}

File: ios/Beacons/BLE/ACABProfile.swift
import CoreBluetooth

/// The ACAB GATT contract. Keep in sync with firmware/lib/acab_core/acab_ble_service.h.
enum ACABProfile {
    static let service    = CBUUID(string: "acab0100-6f75-6973-7079-000000000000")
    static let detections = CBUUID(string: "acab0101-6f75-6973-7079-000000000000")  // notify
    static let config     = CBUUID(string: "acab0102-6f75-6973-7079-000000000000")  // write
    static let status     = CBUUID(string: "acab0103-6f75-6973-7079-000000000000")  // read + notify
    // OTA firmware update: image bytes go up (write-no-response, encrypted), progress
    // notifies come down. Only boards built with in-app OTA expose it; released 1.7
    // boards do NOT, so its presence is a runtime capability check (see BLEManager.otaCapable).
    static let ota        = CBUUID(string: "acab0104-6f75-6973-7079-000000000000")  // write-no-response + notify

    /// What the firmware advertises as.
    static let advertisedName = "ACAB"
}
